queuebacca-core/src/main/java/io/axonif/queuebacca/subscribing/MessageConsumerSelector.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 The Queuebacca Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.axonif.queuebacca.subscribing;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Collections;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+
+import io.axonif.queuebacca.Message;
+import io.axonif.queuebacca.MessageConsumer;
+
+class MessageConsumerSelector<M extends Message> {
+
+    private final NavigableMap<Integer, MessageConsumer<M>> consumers;
+
+    private MessageConsumerSelector(NavigableMap<Integer, MessageConsumer<M>> consumers) {
+        this.consumers = consumers;
```

Comment:
requireNonNull?

Reply:
I could to be defensive. I usually don't if I've made the constructor private and the only value it could be is one provided by a builder.